libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -305,6 +308,18 @@ export function OutQueueManager(
 
     executingQueue = true;
 
+    if (idService && !idServiceCalled) {
+      const xhr = initializeXMLHttpRequest(idService, false, sync);
```

Comment:
This looks good but should we add a connection timeout here like in the other collector requests? In case the ID service doesn't respond so that it doesn't block sending events forever.

Reply:
Added. Removed the status check and added the same `connectionTimeout` proeprty for the value.